src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.AbstractGlobalSuppressMessageCodeAction.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Roslyn.Utilities;
+using static Roslyn.Utilities.PortableShim;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.Suppression
+{
+    internal abstract partial class AbstractSuppressionCodeFixProvider : ISuppressionFixProvider
+    {
+        internal abstract class AbstractGlobalSuppressMessageCodeAction : AbstractSuppressionCodeAction
+        {
+            private readonly Project _project;
+
+            protected AbstractGlobalSuppressMessageCodeAction(AbstractSuppressionCodeFixProvider fixer, Project project)
+                : base(fixer, title: FeaturesResources.SuppressWithGlobalSuppressMessage)
+            {
+                _project = project;
+            }
+
+            protected sealed override async Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(CancellationToken cancellationToken)
+            {
+                var changedSuppressionDocument = await GetChangedSuppressionDocumentAsync(cancellationToken).ConfigureAwait(false);
+                return new CodeActionOperation[]
+                {
+                    new ApplyChangesOperation(changedSuppressionDocument.Project.Solution),
+                    new OpenDocumentOperation(changedSuppressionDocument.Id, activateIfAlreadyOpen: true),
+                    new NavigationOperation(changedSuppressionDocument.Id, position: 0)
+                };
+            }
+
+            protected abstract Task<Document> GetChangedSuppressionDocumentAsync(CancellationToken cancellationToken);
+            public override bool SupportsFixAllOccurrences => true;
+
+            private string GetSuppressionsFilePath(string suppressionsFileName)
+            {
+                if (!string.IsNullOrEmpty(_project.FilePath))
+                {
+                    var fullPath = Path.GetFullPath(_project.FilePath);
+                    var directory = PathUtilities.GetDirectoryName(fullPath);
+                    if (!string.IsNullOrEmpty(directory))
+                    {
+                        var suppressionsFilePath = PathUtilities.CombinePossiblyRelativeAndRelativePaths(directory, suppressionsFileName);
+                        if (!string.IsNullOrEmpty(suppressionsFilePath))
+                        {
+                            return suppressionsFilePath;
+                        }
+                    }
+                }
+
+                return suppressionsFileName;
+            }
+
+            protected async Task<Document> GetOrCreateSuppressionsDocumentAsync(CancellationToken c)
+            {
+                int index = 1;
+                var suppressionsFileName = s_globalSuppressionsFileName + Fixer.DefaultFileExtension;
```

Comment:
minor: This can be moved inside GetSuppressionsFilePath() since you don't need to access any local state to compute this.